File: app/src/main/java/cn/blinkdagger/androidLab/hybird/WebViewFragment.java
package cn.blinkdagger.androidLab.hybird;

import android.content.Intent;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.ProgressBar;

import cn.blinkdagger.androidLab.R;

/**
 * @Author ls
 * @Date 2018/10/23
 * @Description
 * @Version
 */

public class WebViewFragment extends Fragment implements CommonWebView.LoadListener {

    private final static String PARAM_ARGS = "url";

    private CommonWebView dxyWebView;
    private ProgressBar progressBar;
    private FrameLayout mRootView;
    private View mCustomView;


    private int mCurrentProgress;

    private String url;

    public static WebViewFragment newInstance(String url) {
        Bundle args = new Bundle();
        args.putString(PARAM_ARGS, url);
        WebViewFragment fragment = new WebViewFragment();
        fragment.setArguments(args);
        return fragment;
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_dxy_web_view, container, false);

        dxyWebView = rootView.findViewById(R.id.dxy_webView);
        progressBar = rootView.findViewById(R.id.dxy_wv_pb);

        dxyWebView.setListener(this, this);

        return rootView;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initData();

    }

    private void initData() {

        url = getArguments().getString(PARAM_ARGS);
        if (TextUtils.isEmpty(url)) {
            return;
        }
        initWebView();
    }

    @Override
    public void onResume() {
        dxyWebView.onResume();
        super.onResume();
    }

    @Override
    public void onPause() {
        dxyWebView.onPause();
        super.onPause();
    }


    @Override
    public void onDestroy() {
        dxyWebView.onDestroy();
        super.onDestroy();
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent intent) {
        super.onActivityResult(requestCode, resultCode, intent);
        dxyWebView.onActivityResult(requestCode, resultCode, intent);
    }

    private void initWebView() {
        dxyWebView.setUploadFileParam("image/*", "选择图片", true);
        dxyWebView.setListener(this, this);
        dxyWebView.loadUrl(url);
    }

    @Override
    public void onPageStarted(String url, Bitmap favicon) {

    }

    @Override
    public void onPageFinished(String url) {

    }

    @Override
    public void onPageError(int errorCode, String description, String failingUrl) {

    }

    @Override
    public void onExternalPageRequest(String url) {

    }
}

File: app/src/main/java/cn/blinkdagger/androidLab/hybird/WebViewFragment.kt
package cn.blinkdagger.androidLab.hybird

import android.content.Intent
import android.graphics.Bitmap
import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.ProgressBar
import androidx.fragment.app.Fragment
import cn.blinkdagger.androidLab.R

/**
 * @Author ls
 * @Date 2018/10/23
 * @Description
 * @Version
 */
class WebViewFragment : Fragment(), CommonWebView.LoadListener {
    private var mWebView: CommonWebView? = null
    private var progressBar: ProgressBar? = null
    private val mRootView: FrameLayout? = null
    private val mCustomView: View? = null
    private val mCurrentProgress = 0
    private var url: String? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val rootView = inflater.inflate(R.layout.fragment_common_web_view, container, false)
        mWebView = rootView.findViewById(R.id.dxy_webView)
        progressBar = rootView.findViewById(R.id.dxy_wv_pb)
        mWebView?.setListener(this, this)
        return rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initData()
    }

    private fun initData() {
        url = arguments?.getString(PARAM_ARGS)
        if (TextUtils.isEmpty(url)) {
            return
        }
        initWebView()
    }

    override fun onResume() {
        mWebView?.onResume()
        super.onResume()
    }

    override fun onPause() {
        mWebView?.onPause()
        super.onPause()
    }

    override fun onDestroy() {
        mWebView?.onDestroy()
        super.onDestroy()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
        super.onActivityResult(requestCode, resultCode, intent)
        mWebView?.onActivityResult(requestCode, resultCode, intent)
    }

    private fun initWebView() {
        mWebView?.setUploadFileParam("image/*", "选择图片", true)
        mWebView?.setListener(this, this)
        mWebView?.loadUrl(url)
    }

    override fun onPageStarted(url: String?, favicon: Bitmap?) {}
    override fun onPageFinished(url: String?) {}
    override fun onPageError(errorCode: Int, description: String?, failingUrl: String?) {}
    override fun onExternalPageRequest(url: String?) {}

    companion object {
        private const val PARAM_ARGS = "url"
        fun newInstance(url: String?): WebViewFragment {
            val args = Bundle()
            args.putString(PARAM_ARGS, url)
            val fragment = WebViewFragment()
            fragment.arguments = args
            return fragment
        }
    }
}